src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -48,12 +48,26 @@ BoundExpression TranslatePattern(BoundExpression input, BoundPattern pattern)
                         input = _factory.Local(temp);
                         for (int i = 0; i < pat.Subpatterns.Length; i++)
                         {
-                            var subProperty = pat.Subpatterns[i].Property;
+                            // TODO: review and test this code path.
+                            // https://github.com/dotnet/roslyn/issues/9542
+                            var subProperty = (pat.Subpatterns[i].Member as BoundPropertyPatternMember)?.MemberSymbol;
```

Comment:
Another part was removal of conditional access off of subProperty, it is not expected to have null value.